File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let answer = 0;
    let uniqueNum = new Set(nums)
    
    if (nums.length/2 <= uniqueNum.size) {
        answer = nums.length/2;
    }else {
        answer = uniqueNum.size;
    }

    return answer;
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.Arrays;
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int[] pocketmon = Arrays.stream(nums).distinct().toArray();
        
        if(nums.length/2 < pocketmon.length){
            answer = nums.length/2;
        }else{
            answer = pocketmon.length;
        }
        return answer;
    }
}